src/services/document-service.js

```diff
@@ -90,9 +89,28 @@ class DocumentService {
     return this.documentStore.find({ _id: { $in: documentKeys } }, { sort: lastUpdatedFirst, projection: metadataProjection });
   }
 
-  async getDocumentsByTags(searchTags) {
-    const result = await this.documentStore.find({ tags: { $all: searchTags } }, { projection: searchResultsProjection });
-    return result || [];
+  async getDocumentsByTags(searchQuery) {
+    const searchTags = new Set(searchQuery.trim()
+      .split(/\s+/)
+      .map(tag => tag.toLowerCase())
+      .filter(tag => (/^\w{3,30}$/).test(tag)));
+
```

Comment:
We should do an early return here if `searchTags.length === 0`, otherwise we get a 500 cause mongo throws.
Try to search for ". . ."

Reply:
I got that same error, I added a bad request response for strings that have less than 3 chars. But maybe I should do this as well.

Reply:
Yeah true, this doesn't really prevent the ... scenario.

Reply:
Basically any scenario where all tags are shorter than 3 or longer than 30

Reply:
Thanks! this was a bigger issue with escaping the special characters as well. Have a look at the tests please :)

Reply:
I would still do an early return 
```
if (searchTags.length === 0) {
return [];
}
```
and bypass the DB query altogether.

Also, could we maybe rather use an npm package to escape here, since regex has more than just this 2 characters as having special meaning? e.g. maybe https://www.npmjs.com/package/escape-string-regexp ?

Reply:
:+1: Yes, please

Reply:
But I am doing that:

```
if (!searchTags.size) {
      return [];
    }
```

Reply:
Ok cool diffing was not showing this at all to me.

Reply:
I added escaping with the library, thanks! Didn't know about this one and didn't search.

Reply:
Didn't know about it either, just found it. Maybe there are better ones also, dunno, was just a suggestion.